sumologic_test.go

```diff
@@ -145,6 +134,38 @@ func mkTime(secondsAfterBase time.Duration) time.Time {
 	return t.Add(secondsAfterBase * time.Second)
 }
 
+func requestHeadersMatch(expected *RequestData, actual *RequestData) bool {
+	for header, value := range expected.Headers {
+		if !strings.EqualFold(value, actual.Headers[header]) {
+			return false
+		}
+	}
+	return true
+}
+
+func findRequestData(expected []RequestData, actual *RequestData) (*RequestData, error) {
+	for _, req := range expected {
```

Comment:
Should this not perhaps be a `TestSuite` method that calls test assertions? It would simplify checks and error handling and such.

Reply:
Also, we probably want to remove or mark the things we match to deal with duplicates.